src/tests/calculate.test.js

```diff
@@ -0,0 +1,56 @@
+import calculate from '../logic/calculate';
+import '@testing-library/jest-dom';
+
+describe('operate function check', () => {
+  test('AC', () => {
+    const obj = {
+      total: '50',
+      next: '2',
+      operation: '+',
+    };
+    const state = calculate(obj, 'AC');
+    expect(state.total).toBe(0);
+  });
+
+  test('number', () => {
+    const obj = {
+      total: '0',
+      next: null,
+      operation: '+',
+    };
+    const state = calculate(obj, '1');
+    expect(state.next).toBe('1');
+  });
+
+  test('dot', () => {
+    const obj = {
+      total: '3',
+      next: null,
+      operation: null,
+    };
+    const state = calculate(obj, '.');
+    expect(state.total).toBe('3.');
+  });
+
+  test('equal', () => {
+    const obj = {
+      total: '50',
+      next: '2',
+      operation: '+',
+    };
+
+    const state = calculate(obj, '=');
+    expect(state.next).toBe(null);
+  });
+
+  test('+/-', () => {
+    const obj = {
+      total: '50',
+      next: null,
+      operation: null,
+    };
+
+    const state = calculate(obj, '+/-');
+    expect(state.total).toBe('-50');
+  });
```

Comment:
- [ ] @team great work on your tests however you haven't covered all scenarios, kindly check all scenario presented by the if statement and add test for them.

Reply:
- [ ] @team kindly add a test for the app component , this is a project requirement 
![image](https://user-images.githubusercontent.com/61408860/213212360-029c402e-936a-4181-a46a-36be2cc60dd2.png)

Reply:
![Screenshot_2023-01-18-21-13-07-110_com android chrome](https://user-images.githubusercontent.com/113009678/213217884-e9fa59a8-7a58-4d25-8077-d2ed56c26209.jpg)
![Screenshot_2023-01-18-21-13-39-206_com android chrome](https://user-images.githubusercontent.com/113009678/213217903-3eedd78d-b20c-4941-983e-062e7e301858.jpg)

Reply:
I have attached the images of test cases with folder directories.

Reply:
@asifaminisonline I can't see the test for the App component, which I requested for review also for all scenarios in the calculate.js file